autoprotocol/builders.py

```diff
@@ -1632,6 +1619,31 @@ def device_mode_params(
                 "z_drop": z_drop or default_params["z_drop"],
                 "viscosity": viscosity or default_params["viscosity"],
             })
+        LiquidHandleBuilders.validate_device_params(device, device_dict)
```

Comment:
This method is part of the class so you can call self.validate_device_params(device, device_dict)

Reply:
even though device_params is a static method?